src/Sylius/Bundle/CartBundle/EventListener/ClearAdjustmentsOnEmptyCartListener.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CartBundle\EventListener;
+
+use Sylius\Component\Cart\Model\CartInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class ClearAdjustmentsOnEmptyCartListener
+{
+    /**
+     * @param GenericEvent $event
+     */
+    public function clearAdjustmentsOnEmptyCart(GenericEvent $event)
```

Comment:
Hmm... I'm also not sure about this name ;) It doesn't listen on "clearadjustmenronemptycartevent", what is suggested by listener name. It listens to ``sylius.cart_change`` event. I believe, that listener name should be named after it's event (or bunch of events it listens to) and it's **method** should specify what actually does this listener do.
You can move it's method to CartRefreshListener as @stloyd suggested but then maybe it should be named ``CartChangeListener``. Any feedback about that? @michalmarcinkowski @pjedrzejewski?